st2common/st2common/transport/bootstrap_utils.py

```diff
@@ -26,16 +28,47 @@
 from st2common.transport.liveaction import LIVEACTION_XCHG, LIVEACTION_STATUS_MGMT_XCHG
 from st2common.transport.reactor import SENSOR_CUD_XCHG
 from st2common.transport.reactor import TRIGGER_CUD_XCHG, TRIGGER_INSTANCE_XCHG
+from st2common.transport import actionexecutionstate
+from st2common.transport import announcement
+from st2common.transport import execution
+from st2common.transport import liveaction
+from st2common.transport import reactor
 
 LOG = logging.getLogger('st2common.transport.bootstrap')
 
 __all__ = [
-    'register_exchanges'
+    'register_exchanges',
+
+    'EXCHANGES',
+    'QUEUES'
 ]
 
-EXCHANGES = [ACTIONEXECUTIONSTATE_XCHG, ANNOUNCEMENT_XCHG, EXECUTION_XCHG, LIVEACTION_XCHG,
-             LIVEACTION_STATUS_MGMT_XCHG, TRIGGER_CUD_XCHG, TRIGGER_INSTANCE_XCHG,
-             SENSOR_CUD_XCHG]
+# List of exchanges which are pre-declared on service set up.
+EXCHANGES = [
+    ACTIONEXECUTIONSTATE_XCHG,
+    ANNOUNCEMENT_XCHG,
+    EXECUTION_XCHG,
+    LIVEACTION_XCHG,
+    LIVEACTION_STATUS_MGMT_XCHG,
+    TRIGGER_CUD_XCHG,
+    TRIGGER_INSTANCE_XCHG,
+    SENSOR_CUD_XCHG
+]
+
+# List of queues which are pre-declared on service set up.
+# Because of the worker model used, this is required with some non-standard transports such as
+# Redis one. Even though kombu requires queues to be pre-declared upfront in such scenarios,
+# RabbitMQ transport doesn't require that.
+QUEUES = [
+    actionexecutionstate.get_queue(name='st2.preinit', routing_key='init'),
+    announcement.get_queue(name='st2.preinit', routing_key='init'),
+    execution.get_queue(name='st2.preinit', routing_key='init'),
+    liveaction.get_queue(name='st2.preinit', routing_key='init'),
+    liveaction.get_status_management_queue(name='st2.preinit', routing_key='init'),
+    reactor.get_trigger_cud_queue(name='st2.preinit', routing_key='init'),
+    reactor.get_trigger_instances_queue(name='st2.preinit', routing_key='init'),
+    reactor.get_sensor_cud_queue(name='st2.preinit', routing_key='init'),
```

Comment:
Yes, they will never be consumed - in fact only one queue which will never be used and grow will be created. Nothing publishes to them so they never grow.

We only do that, because that's the only way to make sure exchanges are created.

```bash
(virtualenv)vagrant@vagrant-ubuntu-trusty-64:/data/stanley$ sudo rabbitmqctl list_queues
Listing queues ...
init	0
st2.actionrunner.canel	0
st2.actionrunner.req	0
st2.actionrunner.work	0
st2.notifiers.execution.work	4092
st2.notifiers.work	3070
st2.preinit	0
....
```